repos/spack_repo/builtin/packages/moab/package.py

```diff
@@ -94,6 +99,17 @@ class Moab(AutotoolsPackage):
     depends_on("eigen", when="+eigen")
     # FIXME it seems that zoltan needs to be built without fortran
     depends_on("zoltan~fortran", when="+zoltan")
+    depends_on("tempestremap@2.2", when="@master+tempest")
+    depends_on("tempestremap@2.2", when="@5.5.1+tempest")
+    depends_on("tempestremap@2.2", when="@5.5.0+tempest")
+    depends_on("tempestremap@2.1.6", when="@5.4.1+tempest")
+    depends_on("tempestremap@2.1.6", when="@5.4.0+tempest")
+    depends_on("tempestremap@2.1.1", when="@5.3.1+tempest")
+    depends_on("tempestremap@2.1.0", when="@5.3.0+tempest")
+    depends_on("tempestremap@2.0.5", when="@5.2.1+tempest")
+    depends_on("tempestremap@2.0.3", when="@5.2.0+tempest")
+    depends_on("tempestremap@2.0.2", when="@5.1.0+tempest")
+    depends_on("eigen", when="+tempest")
```

Comment:
Is it necessary to pin `tempestremap` versions this much? 

Also, you can write:
```suggestion
    with when("+tempest"):
        depends_on("tempestremap@2.2", when="@5.5:")
        depends_on("tempestremap@2.1.6", when="@5.4")
        depends_on("tempestremap@2.1.1", when="@5.3.1")
        depends_on("tempestremap@2.1.0", when="@5.3.0")
        depends_on("tempestremap@2.0.5", when="@5.2.1")
        depends_on("tempestremap@2.0.3", when="@5.2.0")
        depends_on("tempestremap@2.0.2", when="@5.1.0")
        depends_on("eigen")
```

Reply:
> Is it necessary to pin tempestremap versions this much?

Yes, each `moab` version only works with one specific `tempestremap` version.  These pins are required.